pennylane/grouping/group_observables.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the high-level Pauli word partitioning functionality used in measurement optimization.
+"""
+
+from pennylane.wires import Wires
+from pennylane.grouping.utils import (
+    convert_observables_to_binary_matrix,
+    binary_to_pauli,
+    are_identical_pauli_words,
+    get_qwc_compliment_adj_matrix,
+)
+from pennylane.grouping.graph_colouring import largest_first, recursive_largest_first
+import numpy as np
+
+GROUPING_TYPES = ["qwc", "commuting", "anticommuting"]
```

Comment:
So far, only QWC is implemented; it might be a better to introduce references to the other schemes when they're ready. After all, we don't want users to see a scheme available, only to try it out and find out it actually isnt.

Also, use a `frozenset`, since immutability is the best thing in programming since sliced bread.

Reply:
That's fair. Then again, even without the commuting and anticommuting partitioning schemes ready for measurement optimization, PauliGroupingStrategy may find some utility outside the context of measurement optimization. For instance, partitioning Paulis to anticommuting sets has found some utility in [ansatz construction](https://arxiv.org/abs/2002.05701). Similarly, the commutativity of Pauli terms in unitary ansatz [can play an important role](https://arxiv.org/pdf/1910.10329.pdf), and one may find access to partitioning info useful. 

Also, noted on the use of `frozenset`. :smile: